extensions-core/kinesis-indexing-service/src/main/java/org/apache/druid/indexing/kinesis/KinesisRecordSupplier.java

```diff
@@ -780,15 +780,14 @@ private String getSequenceNumberInternal(StreamPartition<String> partition, Shar
   private String getSequenceNumberInternal(StreamPartition<String> partition, String shardIterator)
   {
     long timeoutMillis = System.currentTimeMillis() + fetchSequenceNumberTimeout;
+    GetRecordsResult recordsResult = null;
```

Comment:
I want to access recordsResult outside of the while loop (so that I can print the millisBehindLatest in the warn logging line 834). Hence, I moved variable declaration of recordsResult to outside the while loop and we can check for null in the warn logging. If it is not null (meaning it was assign in the while loop when we getRecords from Kinesis) then we can log the millisBehindLatest. This will help us be sure that timeout is due to no data or not

Reply:
Got it. Thanks.